File: 程式設計：資料結構答案檔/Go/201.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func requestQuests(input string) ([]int, error) {
	tasks := []int{}
	for _, part := range strings.Split(input, ",") {
		taskReward, err := strconv.Atoi(part)
		if err != nil {
			return nil, fmt.Errorf("error")
		}
		if taskReward < 0 {
			return nil, fmt.Errorf("error")
		}
		tasks = append(tasks, taskReward)
	}
	return tasks, nil
}

func calculateTotalReward(tasks []int) int {
	totalReward := 0
	morningTasks := 3
	noonTasks := 2
	nightTasks := 1

	i := 0
	for i < len(tasks) {
		for j := 0; j < morningTasks && i < len(tasks); j++ {
			totalReward += tasks[i] * 3
			i++
		}
		for j := 0; j < noonTasks && i < len(tasks); j++ {
			totalReward += tasks[i] * 2
			i++
		}
		for j := 0; j < nightTasks && i < len(tasks); j++ {
			totalReward += tasks[i]
			i++
		}
	}
	return totalReward
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	input, _ := reader.ReadString('\n')
	input = strings.TrimSpace(input)

	tasks, err := requestQuests(input)
	if err != nil {
		fmt.Println(err)
		return
	}

	totalReward := calculateTotalReward(tasks)
	fmt.Println(totalReward)
}

File: 程式設計：資料結構答案檔/Python/201.py
def request_quests(input_str):
    tasks = []
    for part in input_str.split(','):
        try:
            task_reward = int(part)
            if task_reward < 0:
                print("error")
                return None
            tasks.append(task_reward)
        except ValueError:
            print("error")
            return None
    return tasks

def calculate_total_reward(tasks):
    total_reward = 0
    morning_tasks = 3
    noon_tasks = 2
    night_tasks = 1

    i = 0
    while i < len(tasks):
        for _ in range(morning_tasks):
            if i < len(tasks):
                total_reward += tasks[i] * 3
                i += 1
        for _ in range(noon_tasks):
            if i < len(tasks):
                total_reward += tasks[i] * 2
                i += 1
        for _ in range(night_tasks):
            if i < len(tasks):
                total_reward += tasks[i]
                i += 1
                
    return total_reward

input_str = input()
tasks = request_quests(input_str)
if tasks is not None:
    total_reward = calculate_total_reward(tasks)
    print(total_reward)
